crates/pallets/starknet/src/tests/query_tx.rs

```diff
@@ -97,3 +98,23 @@ fn query_tx_should_not_be_executable() {
         );
     });
 }
+
+#[test]
+fn estimate_fee_skips_validation_for_invalid_tx() {
+    new_test_ext::<MockRuntime>().execute_with(|| {
+        basic_test_setup(2);
+
+        let invalid_transaction = get_invoke_dummy(Starknet::chain_id(), Nonce(StarkFelt::ZERO));
+        let tx = AccountTransaction::Invoke(invalid_transaction.clone());
+
+        let tx_vec = vec![tx.clone()];
+
+        let fee_estimation_result =
+            Starknet::estimate_fee(tx_vec.clone(), &SimulationFlags { validate: true, charge_fee: true });
```

Comment:
You are not skiping validation here: `validate: true`

Reply:
ideally we want something like this:

``rust
let fee_estimation_result =
            Starknet::estimate_fee(tx_vec.clone(), &SimulationFlags { validate: true, charge_fee: true });
assert!(fee_estimation_result.is_err(), "Fee estimation succeded");
let fee_estimation_result =
            Starknet::estimate_fee(tx_vec.clone(), &SimulationFlags { validate: false, charge_fee: true });
assert!(fee_estimation_result.is_ok(), "Fee estimation failed unexpectedly");
```

to make sure that the skipping of the validation did had the intended effect